src/java/uk/ac/rdg/resc/edal/cdm/DataChunk.java

```diff
@@ -118,10 +118,11 @@ public Index getIndex() {
      * @return the data value, or {@link Float#NaN} if this is a missing value
      */
     public float readFloatValue(Index index) {
-        double val = arr.getDouble(index);
-        if (this.needsEnhance)
-            return (float) this.var.convertScaleOffsetMissing(val);
-        else
-            return (float) val;
+        double val = arr.getFloat(index);
```

Comment:
Okay, `VariableDS.convertScaleOffsetMissing()` and `VariableDS.isMissing()` only accept `double`, which is why that's `val`'s type. The implicit cast here is a bit confusing though. Also, for the record, assuming that `arr` is an `ArrayFloat`, the old statement:
```java
double val = arr.getDouble(index);
```
has the same effect as the new statement:
```java
double val = arr.getFloat(index);
```
The old statement is arguably more clear.